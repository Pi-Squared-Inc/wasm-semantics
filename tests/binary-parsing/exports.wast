(module $mymodule
  (export "mytable" (table 0))
  (export "mymem" (memory 0))
  (export "myglobal" (global 1))
  (export "myfunc" (func 2))

  (global i32 (i32.const 1))
  (global (mut i64) (i64.const 2))
  (memory 17 20)
  (func (param i32) (result i32)
    (return
      (i32.const 1)
    )
  )
  (func (param i32) (result i32)
    (return
      (i32.const 1)
    )
  )
  (func (param i32) (result i32)
    (return
      (i32.const 1)
    )
  )
  (table 1 funcref)
)
