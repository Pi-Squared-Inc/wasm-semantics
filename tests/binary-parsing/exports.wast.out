<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: #table (... limits: #limitsMin ( 1 ) , type: funcref , metadata:  )  .EmptyStmts , mems: #memory (... limits: #limits ( 17 , 20 ) , metadata:  )  .EmptyStmts , globals: #global (... type: const i32 , init: i32 . const 1  .EmptyStmts , metadata:  )  #global (... type: var i64 , init: i64 . const 2  .EmptyStmts , metadata:  )  .EmptyStmts , elem: .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: #export (... name: "mytable" , index: 0 )  #export (... name: "mymem" , index: 0 )  #export (... name: "myglobal" , index: 1 )  #export (... name: "myfunc" , index: 2 )  .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
